.github/workflows/update-publications.yaml

```diff
@@ -14,47 +17,53 @@ jobs:
   update-publications:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-
+      - uses: actions/checkout@v4
       - uses: r-lib/actions/setup-r@v2
         with:
-          r-version: '4.1.3'
-
-      - name: Query dependencies
-        run: |
-          install.packages('remotes')
-          install.packages("synapser")
-          saveRDS(remotes::dev_package_deps(dependencies = TRUE), "depends.Rds", version = 2)
-        shell: Rscript {0}
+          r-version: '4.3.1'
 
       - name: Install system dependencies
         run: |
           export DEBIAN_FRONTEND=noninteractive
-          sudo apt-get -y update
-          sudo apt-get install -y make python2-minimal pandoc pandoc-citeproc git-core libv8-dev libxml2-dev libcurl4-openssl-dev libssl-dev libssh2-1-dev zlib1g-dev libgit2-dev
+          sudo apt-get update && \
+            sudo apt-get -y install libpng-dev \
+                pandoc \
+                git-core \
+                libv8-dev \
+                libxml2-dev \
+                libcurl4-openssl-dev \
+                libssl-dev \
+                libssh2-1-dev \
+                zlib1g-dev \
+                libgit2-dev \
+                python3 \
+                python3-pip \
+                python3-dev \
+                python3-venv
+                # pandoc-citeproc
 
       - name: Install dependencies
         run: |
-          library(remotes)
-          deps <- readRDS('depends.Rds')
-          deps[['installed']] <- vapply(deps[['package']], remotes:::local_sha, character(1))
-          update(deps)
-        shell: Rscript {0}
-
-      - name: Install Miniconda
-        run: |
-          Rscript -e "remotes::install_github('rstudio/reticulate')"
-          Rscript -e "reticulate::install_miniconda()"
+          install.packages('remotes')
+          remotes::install_cran('rentrez')
+          remotes::install_cran('librarian')
+          remotes::install_version('rjson', version='0.2.21')
+          remotes::install_version('reticulate', version='1.28')
+          reticulate::install_miniconda()
+          reticulate::conda_remove("r-reticulate")
+          # Create a new Conda environment with Python 3.10
+          reticulate::conda_create("r-reticulate", python_version = "3.10")
```

Comment:
The default python is 3.8, and future versions of the synapse python client will not work with that python version.